app/services/addresses/manage_order_address_service.rb

```diff
@@ -0,0 +1,32 @@
+class Addresses::ManageOrderAddressService
+  def initialize(order, params, use_billing)
+    @order = order
+    @params = params
+    @use_billing = use_billing
+  end
+
+  def call
+    Address.casts.keys.each do |cast|
+      address = get_address(cast.to_sym)
+      params = address_params(set_type(cast.to_sym))
+      address.exists? ? address.update(params) : address.create(params)
+    end
+  end
+
+  private
+
+  def get_address(cast)
+    @order.addresses.public_send(cast)
+  end
+
+  def set_type(cast)
+    case cast
+    when :billing then :billing
```

Comment:
what is :billing and :shipping? You should connect them with constants/enumbs inside Address